File: Resource/51900286-51900712-51900718-51900699/source demo/Window Form/51900286-51900712-51900718-51900699/RestaurentDTO/LichHen.cs
using System;
using System.Data;

namespace RestaurentDTO
{
    public class LichHen
    {
        private string _maLichHen;
        private string _nhuCau;
        private int _soLuongKhach;
        private string _thoiGian;
        private string _ngayHen;
        private string _tinhTrang;
        private string _maKhachHang;
        private string _maNhanVien;
        private string _maBanAn;

        public LichHen(string maLichHen, string nhuCau, int soLuongKhach, string thoiGian, string ngayHen, string tinhTrang, string maKhachHang, string maNhanVien, string maBanAn)
        {
            _maLichHen = maLichHen;
            _nhuCau = nhuCau;
            _soLuongKhach = soLuongKhach;
            _thoiGian = thoiGian;
            _ngayHen = ngayHen;
            _tinhTrang = tinhTrang;
            _maKhachHang = maKhachHang;
            _maNhanVien = maNhanVien;
            _maBanAn = maBanAn;
        }

        public LichHen()
        {
            _maLichHen = "";
            _nhuCau = "";
            _soLuongKhach = 0;
            _thoiGian = "";
            _ngayHen = "";
            _tinhTrang = "0";
            _maKhachHang = "";
            _maNhanVien = "";
            _maBanAn = "";
        }

        public LichHen(DataRow dataRow)
        {
            _maLichHen = dataRow["MaLichHen"].ToString();
            _nhuCau = dataRow["NhuCau"].ToString();
            _soLuongKhach = (int)dataRow["SoLuongKhach"];
            _thoiGian = dataRow["ThoiGian"].ToString();
            _ngayHen = dataRow["NgayHen"].ToString();
            _tinhTrang = dataRow["TinhTrang"].ToString();
            _maKhachHang = dataRow["MaKhachHang"].ToString();
            _maNhanVien = dataRow["MaNhanVien"].ToString();
            _maBanAn = dataRow["MaBanAn"].ToString();
        }



        public string MaLichHen { get => _maLichHen; set => _maLichHen = value; }
        public string NhuCau { get => _nhuCau; set => _nhuCau = value; }
        public int SoLuongKhach { get => _soLuongKhach; set => _soLuongKhach = value; }
        public string ThoiGian { get => _thoiGian; set => _thoiGian = value; }
        public string NgayHen { get => _ngayHen; set => _ngayHen = value; }
        public string TinhTrang { get => _tinhTrang; set => _tinhTrang = value; }
        public string MaKhachHang { get => _maKhachHang; set => _maKhachHang = value; }
        public string MaNhanVien { get => _maNhanVien; set => _maNhanVien = value; }
        public string MaBanAn { get => _maBanAn; set => _maBanAn = value; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/LichHen.java
package tdtu.advanced.java.thinh68.models;

import java.sql.Date;
import java.sql.Timestamp;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;


@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name="lich_hen")
public  class LichHen {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    @Column(name = "ma_lich_hen")
    private Long maLichHen;
    
    @Column(name = "nhu_cau")
    private String nhuCau;
    
    @Column(name = "so_luong_khach")
    private int soLuongKhach;
    
    @Column(name = "thoi_gian")
    public Timestamp thoiGian;
    
    @Column(name = "ngay_hen")
    private Date ngayHen;
    
    @Column(name = "tinh_trang")
    private int tinhTrang;
    
    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lich_hen_ban_an_foreign_key", referencedColumnName="ma_ban_an")
    private BanAn banAn;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lich_hen_khach_hang_foreign_key", referencedColumnName="ma_khach_hang", nullable = false)
    private KhachHang khachHang;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lich_hen_nhan_vien_foreign_key", referencedColumnName="ma_nhan_vien")
    private NhanVien nhanVien;
}